### ISSUE TITLE
Error : License signature is invalid

### ISSUE THREAD

**Issue by Youenlb** [readium/readium-lcp-server#317]
Hi! 
I need to set up a solution to encrypt a pdf book. Then to decrypt it using the associated Thorium license.
For that I followed these steps : 
- Installation of the lcp and lsd server, and the lcpencrypt tool
- Launching the 2 servers 
- Encrypting the publication with the lcpencrypt tool
-Doing API call with CURL in php, to add the encrypted publication on the lcp server, and generating the .lcpl license

However when I open this license, I get the error "License signature is invalid" in Thorium application.

My question is the following: for the generation of a license that can be opened on Thorium, it is necessary in this case to have a root certicate by readium, so that the license is recognized?
Or does the error come from my API calls, especially when I give my encryption key

**Comment by danielweck** [comment]
Hello, the "License signature is invalid" in Thorium means that the signature found in your LCP license JSON (property value `signature.value`) does not match the signature calculated for the canonical form of the LCPL file (i.e. ordered JSON keys, whitespace removed, signature JSON property removed, etc.). The RSA public key (or ECDSA in the case of LCP 1.0 production profile) that is used to check the signature is extracted from the LCP license's x509 certificate (i.e. the "provider" cert configured in your LCP server instance).
When the "License signature is invalid" error is emitted, we know that at this stage in the verification sequence your "provider" certificate's signature has been successfully validated against the public key of EDRLab's CA "root" certificate (and also that your provider cert is not revoked or expired).
Therefore I conclude that your generated LCP license is somehow modified after it is signed.

**Comment by Youenlb** [comment]
Ok, thanks a lot for your feedback! 
However, when I try to regenerate a license, I get an error about the status of my profile ("basic"), I managed to solve this problem by removing the profile attribute from my yaml file. However, when I try to regenerate my license, I get the error "incompatible LCP profile; error generating a user key" in return of my license generation request. So my question is, is the server yaml configuration different depending on the profile status ?

**Comment by llemeurfr** [comment]
Please keep in your yaml config file the line
`profile: basic`

And check if you have used the EDRLab test certificates in your config, lines should be:
```
certificate:
    cert: (path)/cert-edrlab-test.pem
    private_key: (path)/privkey-edrlab-test.pem
```

**Comment by Youenlb** [comment]
Yes, indeed, I had corrected it now I can generate my license, however I have the error: "DAISY3 OPF package XML file or DAISY2 NCC cannot be found" in Thorium and I can't find the XML package to download to solve this problem.
(Sorry for the multiple questions).

**Comment by llemeurfr** [comment]
This is a very different issue, which has to do with the DAISY format vs LCP. 
Are you trying to encrypt a DAISY publication? 
There has been no work done for protecting DAISY files with the LCP DRM. It may be developed later, we don't know if it is interesting.

**Comment by Youenlb** [comment]
Yes, I find it strange to have this error because it is a PDF that I encrypted with the tool you propose. Then I store the encrypted publication in a zip with its manifest.json and the cover page.

**Comment by Youenlb** [comment]
Yes indeed I did not follow exactly these steps, I will make the changes to see if the error comes from the zip. Thanks !

**Comment by llemeurfr** [comment]
I'll close this issue now. Please open a new one on the proper subject of the issue if you still have problems
